### PR TITLE
fix(msteams): use correct team id

### LINKED ISSUE #66130: Microsoft Teams Integration Problem
### Environment

SaaS (https://sentry.io/)

### Steps to Reproduce

1. Following your MS teams instructions, I have added the the Sentry bot to a team.
2. I have completed the integration with the Complete setup message I get from the bot:
image
3. Afterwards I try to set Alerts, but I get no MSTeams possibility in the create rule interface:
image

### Expected Result

I expect the possibility to select the MSTeams action in the dropdown, like shown in the documentation: https://docs.sentry.io/product/integrations/notification-incidents/msteams/

### Actual Result

Actual result:
image

### Product Area

Settings - Integrations

### Link

_No response_

### DSN

_No response_

### Version

_No response_
```[tasklist]
### Tasks
```

### PR DESCRIPTION
**Pull request by ykamo001**
Use the correct id when leveraging the team id
https://learn.microsoft.com/en-us/microsoftteams/platform/resources/bot-v3/bots-notifications#schema-example-bot-added-to-team

https://learn.microsoft.com/en-us/microsoftteams/platform/bots/how-to/conversations/subscribe-to-conversation-events?tabs=dotnet#installation-update-event

### CODE REVIEW

File: src/sentry/integrations/msteams/webhook.py [hunk 1]
```diff
@@ -264,18 +264,11 @@ def _get_team_installation_request_data(cls, data: dict[str, Any]) -> dict:
...
-        if team_id is None:
-            logger.info(
-                "sentry.integrations.msteams.webhooks: New team info data does not have aadGroupId",
-                extra={"data": data},
-            )
-            fallback_id = new_team_info["id"]
-            team_id = fallback_id
 
+        new_team_info = channel_data["team"]
+        team_id = new_team_info["id"]
```
Comment (cathteng) [inline src/sentry/integrations/msteams/webhook.py:269]: this will always be there?

Comment (ykamo001) [reply src/sentry/integrations/msteams/webhook.py:269]: it should based on the docs!

Comment (ykamo001) [reply src/sentry/integrations/msteams/webhook.py:269]: if not, we will raise and catch the exception, responding back with a 400

**Review by cathteng — APPROVED** [review]
i read the docs, and this looks good

**Pull request merged**